packages/salesforcedx-vscode-core/src/util/authInfo.ts

```diff
@@ -80,3 +73,33 @@ export class OrgAuthInfo {
     }
   }
 }
+
+enum VSCodeWindowTypeEnum {
+  Error = 1,
+  Informational = 2,
+  Warning = 3
+}
+function displayMessage(
```

Comment:
Most of the functions duplicated throughout the core extension for displaying messages end up displaying only a single type of message (warning/error/informational). Since OrgAuthInfo needs to display each type of message, I'd opted for a single function that writes to the output window and optionally displays a messages. This was preemptively done like this because getDefaultUsernameOrAlias ends up getting called twice each time a project is loaded, once by  the OrgPicker and once by the WorkspaceOrgType. Some refactoring needs to get done to reconcile those but there probably needs to be a boolean plumbed through the getDefaultUsernameOrAlias call so the message only gets displayed once. Right now, it's odd though, only one notification gets popped up in spite of there being two calls and two messages in the output window.

Reply:
Is this generic enough to live outside of authInfo in another util?

Reply:
Possibly but due to time constraints it's just going to live here for the moment.